index.go

```diff
@@ -75,7 +75,7 @@ func (o *IndexDef) changeColumn(originalName, newName string) {
 }
 
 func (o *IndexDef) hasPrefix(columns []string) bool {
-	if len(o.Columns) < len(columns) {
+	if len(o.Columns) > len(columns) {
```

Comment:
You should not modify this condition. If `o.Columns` is longger than `colums`, it may has a prefix of `colums`.
You should modify the `for` iteration.